django/db/models/base.py

```diff
@@ -1923,6 +1924,35 @@ def _check_constraints(cls, databases):
                         id='models.W038',
                     )
                 )
+
+            def fetch_check_constraint_check_fields(check):
+                fields = set()
+                if isinstance(check, RawSQL):
+                    return fields
+                if isinstance(check, ExpressionWrapper):
+                    children = check.get_source_expressions()
+                elif isinstance(check, Q):
+                    children = check.children
+                for child in children:
+                    if isinstance(child, Q):
+                        fields.update(fetch_check_constraint_check_fields(child))
+                    else:
+                        fields.add(child[0].split(LOOKUP_SEP)[0])
+                        if isinstance(child[1], F):
+                            fields.add(child[1].name)
+                return fields
+
+            for constraint in cls._meta.constraints:
+                if isinstance(constraint, UniqueConstraint):
+                    errors.extend(cls._check_local_fields(constraint.fields, "unique_constraint"))
+                elif isinstance(constraint, CheckConstraint):
+                    errors.extend(
+                        cls._check_local_fields(
+                            fetch_check_constraint_check_fields(constraint.check),
+                            "check_constraint"
```

Comment:
I think we can do better here:

```suggestion
                            "CheckConstraint '{}'".format(constraint.name)
```

This would make the message look like:
```
"CheckConstraint 'constraint_name' refers to the nonexistent field 'missing_field'."
```

The same applies for `UniqueConstraint`.

Reply:
@Ian-Foote Thanks. But by your suggested change, the error message would be like:
`"'CheckConstraint 'constraint_name'' refers to the nonexistent field 'missing_field'."`

Reply:
Ah... That's irritating. I still think it would be better to improve the error message, but it looks like it'll require slightly deeper changes.